netci.groovy

```diff
@@ -2,16 +2,71 @@
 import jobs.generation.*;
 
 // The input project name
-def project = GithubProject
+project = GithubProject
 
 // The input branch name (e.g. master)
-def branch = GithubBranchName
+branch = GithubBranchName
 
 // What this repo is using for its machine images at the current time
-def imageVersionMap = ['Windows_NT':'latest-dev15-5',
-                       'OSX10.13':'latest-or-auto',
-                       'Ubuntu14.04':'latest-or-auto',
-                       'Ubuntu16.04':'20170731']
+imageVersionMap = ['Windows_NT':'latest-dev15-5',
+                    'OSX10.13':'latest-or-auto',
+                    'Ubuntu14.04':'latest-or-auto',
+                    'Ubuntu16.04':'20170731',
+                    'RHEL7.2' : 'latest']
+
+def CreateJob(script, runtime, osName, isPR, shouldSkipTestsWhenResultsNotFound=false, isSourceBuild = false) {
```

Comment:
You'll need to pass in the "project and branch" variables here
Also, the "job" method is not going to be available here.  You'll need to pass in the dsl factory when you call CreateJob.  For example, here is similar usage.

https://github.com/dotnet/dotnet-ci/blob/master/src/jobs/generation/Utilities.groovy#L1167

Reply:
So as it is right now, project, branch, and job will just evaluate to null? I can probably capture them in a class right? To avoid the super long method call.

Reply:
I think you might be able to use "this.project" in the method

Reply:
(where this == the dslFactory parameter)

Reply:
What's the relationship between the dslFactory, project and branch, and the netci.groovy script class? Are they just local vars in the run method?

Reply:
So I ran one of the generated test jobs and it seems to just work: https://ci2.dot.net/job/Microsoft_msbuild/job/vs15.7/job/GenPRTest/job/innerloop_Windows_NT_CoreCLR/1/console

Does this mean it's all good? :)

Reply:
project and branch are just local variables to the script
dslFactory is passed in as the "this" pointer to the script, which is the class context that can be used to invoke the DSL methods, e.g.:

```
foo(factory) {
    factory.job {
    }
}

foo(this)
```